context/app/static/js/components/Detail/SummaryData/SummaryData.spec.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import { render } from 'test-utils/functions';
+import SummaryData from './SummaryData';
+
+test('json button exists and has href', () => {
+  const { getByRole } = render(
+    <SummaryData entity_type="Fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  expect(getByRole('link')).toBeInTheDocument();
+  expect(getByRole('link')).not.toBeEmptyDOMElement();
+  expect(getByRole('link')).toHaveAttribute('href', `/browse/fake/fakeuuid.json`);
+});
+
+test('dataset displays properly', () => {
+  const { getByText, getByTestId } = render(
+    <SummaryData entity_type="Dataset" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  const textToTest = ['QA', 'Public Access'];
+  textToTest.forEach((text) => expect(getByText(text)).toBeInTheDocument());
+  expect(getByTestId('status-svg-icon')).toBeInTheDocument();
+});
+
+test('non-dataset displays properly', () => {
+  const { queryByTestId } = render(
+    <SummaryData entity_type="fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  expect(queryByTestId('status-svg-icon')).toBeNull();
+});
+
+test('children display when provided', () => {
+  const { getByText, getByTestId } = render(
+    <SummaryData entity_type="fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public">
+      <>
+        <div>child 1</div>
+        <div>child 2</div>
+      </>
+    </SummaryData>,
+  );
+  expect(getByTestId('summary-data-parent')).toBeInTheDocument();
+  expect(getByTestId('summary-data-parent')).not.toBeEmptyDOMElement();
+
+  const textToTest = ['child 1', 'child 2'];
+  textToTest.forEach((text) => expect(getByText(text)).toBeInTheDocument());
+});
+
+test('children do not display when undefined', () => {
+  const { queryByText, queryByTestId } = render(
+    <SummaryData entity_type="fake" uuid="fakeuuid" status="QA" mapped_data_access_level="Public" />,
+  );
+  expect(queryByTestId('summary-data-parent')).toBeNull();
+
+  const textToTest = ['child 1', 'child 2'];
+  textToTest.forEach((text) => expect(queryByText(text)).toBeNull());
```

Comment:
```suggestion
```
Maybe there's a way to check that it's empty? There's nothing special about 'child 1'... Could just as well say 'arbitrary-string-we-haven't-mentioned-anywhere'.

Reply:
@mccalluc I am checking its emptiness directly above. While there isn't anything special about 'child 1', I am checking that text is being displayed. I guess I don't understand the issue?

Reply:
@mccalluc Sorry, I misunderstood the test on which you commented. You are correct, I removed the assertion.